src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -979,3 +975,33 @@ ColumnType(Oid relationId, char *columnName)
 
 	return columnType;
 }
+
+
+/*
+ * Check that the current replication factor setting is compatible with the
+ * replication model of relationId, if valid. If InvalidOid, check that the
+ * global replication model setting instead. Errors out if an invalid state
+ * is detected.
+ */
+void
+EnsureReplicationSettings(Oid relationId)
+{
+	char replicationModel = (char) ReplicationModel;
+	char *extraHint = " or setting \"citus.replication_model\" to \"statement\"";
```

Comment:
Then I can't distinguish between the GUC being wrong and an attempt to create an empty shard for a table with streaming replication.